libcudacxx/.upstream-tests/test/cuda/ptx/ptx.mbarrier.wait.compile.pass.cpp

```diff
@@ -0,0 +1,172 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+// UNSUPPORTED: libcpp-has-no-threads
+
+// <cuda/ptx>
+
+#include <cuda/ptx>
+#include <cuda/std/utility>
+
+template <typename ... _Ty>
+__device__ inline bool __unused(_Ty...) { return true; }
+
+__global__ void test_compilation() {
+  using cuda::ptx::sem_release;
+  using cuda::ptx::space_cluster;
+  using cuda::ptx::space_shared;
+  using cuda::ptx::scope_cluster;
+  using cuda::ptx::scope_cta;
+
+  __shared__ uint64_t bar;
+  bar = 1;
+  uint64_t * addr = &bar;
+  uint64_t state = 1;
+  uint32_t phaseParity = 1;
+  uint32_t suspendTimeHint = 1;
+  bool waitComplete = true;
+
+  int thread_filter = 1024;
+
+#if __cccl_ptx_isa >= 700
+  NV_IF_TARGET(NV_PROVIDES_SM_80, (
+    if (threadIdx.x > thread_filter++) {
```

Comment:
These weird if statements have been successful in the past to prevent `ptxas` from segfaulting. Not anymore though. I am still working on a fix.